airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -237,13 +247,20 @@ public AirbyteCatalog discover(final JsonNode config) throws Exception {
 
     if (PostgresUtils.isCdc(config)) {
       final List<AirbyteStream> streams = catalog.getStreams().stream()
-          .map(PostgresCdcCatalogHelper::overrideSyncModes)
-          .map(PostgresCdcCatalogHelper::removeIncrementalWithoutPk)
-          .map(PostgresCdcCatalogHelper::setIncrementalToSourceDefined)
-          .map(PostgresCdcCatalogHelper::addCdcMetadataColumns)
+          .map(PostgresCatalogHelper::overrideSyncModes)
+          .map(PostgresCatalogHelper::removeIncrementalWithoutPk)
+          .map(PostgresCatalogHelper::setIncrementalToSourceDefined)
+          .map(PostgresCatalogHelper::addCdcMetadataColumns)
           // If we're in CDC mode and a stream is not in the publication, the user should only be able to sync
           // this in FULL_REFRESH mode
-          .map(stream -> PostgresCdcCatalogHelper.setFullRefreshForNonPublicationStreams(stream, publicizedTablesInCdc))
+          .map(stream -> PostgresCatalogHelper.setFullRefreshForNonPublicationStreams(stream, publicizedTablesInCdc))
+          .collect(toList());
+
+      catalog.setStreams(streams);
+    } else if (PostgresUtils.isXmin(config)) {
```

Comment:
This means that no existing incremental connection will automatically switch to xmin, right?

Reply:
That's correct. It's a new sync mode that users will have to choose. There isn't currently a migration path